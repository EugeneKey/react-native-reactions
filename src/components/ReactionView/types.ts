import type {
  StyleProp,
  TextStyle,
  TouchableOpacityProps,
  ViewStyle,
} from 'react-native';

export interface ReactionViewProps {
  children: JSX.Element;
  items?: EmojiItemProp[];
  onTap?: (e: EmojiItemProp | undefined) => void;
  cardStyle?: StyleProp<ViewStyle>;
  emojiStyle?: StyleProp<TextStyle>;
  touchableProps?: TouchableOpacityProps;
  emojiKey?: string;
  iconSize?: number;
}

export interface EmojiItemProp {
  // emojiData object is indexed with a Key it will return a value of any type.
  [key: string]: any;
  id: number;
  emoji: React.ReactNode | string | number;
  title: string;
}
