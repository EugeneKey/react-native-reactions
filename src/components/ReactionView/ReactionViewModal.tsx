import React, { useRef } from 'react';
import { TouchableOpacity } from 'react-native';
import { reactionModalRef } from '../ReactionModal';
import type { ReactionViewProps } from './types';

const ReactionViewModal = ({
  children,
  touchableProps,
  ...props
}: ReactionViewProps) => {
  const rootRef = useRef<TouchableOpacity>(null);
  const contentHeightRef = useRef<number>(0);
  const contentyRef = useRef<number>(0);

  const onPressHandler = () => {
    rootRef?.current &&
      rootRef?.current.measureInWindow(
        (x: number, y: number, width: number) => {
          reactionModalRef.current &&
            reactionModalRef.current?.show({
              x,
              y,
              width,
              contentHeight: contentHeightRef.current,
              ...props,
            });
          contentyRef.current = y;
        }
      );
  };

  return (
    <TouchableOpacity
      ref={rootRef}
      disabled={
        children?.props?.hasOwnProperty('onPress') ||
        children?.props?.hasOwnProperty('onLongPress')
      }
      {...touchableProps}
      onLayout={event => {
        const { height } = event.nativeEvent.layout;
        contentHeightRef.current = height;
      }}
      onLongPress={onPressHandler}
      onPress={onPressHandler}>
      {React.isValidElement(children) &&
        React.cloneElement(children as React.ReactElement, {
          onLongPress: onPressHandler,
          onPress: onPressHandler,
        })}
    </TouchableOpacity>
  );
};

export default ReactionViewModal;
