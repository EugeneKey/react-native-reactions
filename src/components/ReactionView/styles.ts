import { StyleSheet } from 'react-native';
import { Colors, moderateScale, verticalScale } from '../../theme';

const styles = StyleSheet.create({
  main: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  container: {
    position: 'absolute',
  },
  root: {
    zIndex: 11,
  },
  subContainer: {
    zIndex: 10,
  },
  emojiBox: {
    flexDirection: 'row',
    bottom: moderateScale(5),
    padding: moderateScale(9),
    backgroundColor: Colors.blue,
    borderRadius: 30,
    shadowColor: Colors.black,
    shadowOffset: {
      width: moderateScale(3),
      height: verticalScale(2),
    },
    shadowOpacity: 0.5,
    shadowRadius: 1,
  },
});

export default styles;
