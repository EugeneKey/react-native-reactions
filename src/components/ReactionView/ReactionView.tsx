import React from 'react';
import { SafeAreaView, Text } from 'react-native';
import styles from './styles';
import type { ReactionViewProps } from './types';

const ReactionsView = ({ items = [] }: ReactionViewProps) => {
  return (
    <SafeAreaView style={styles.container}>
      <Text>{items?.length}</Text>
    </SafeAreaView>
  );
};

export default ReactionsView;
