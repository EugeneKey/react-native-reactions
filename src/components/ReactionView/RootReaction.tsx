import React from 'react';
import ReactionView from './ReactionView';
import ReactionViewModal from './ReactionViewModal';
import type { ReactionViewProps } from './types';
import { GlobalConstants } from '../../constants';

const RootReaction = (props: ReactionViewProps) => {
  const { type } = props;
  return type === GlobalConstants.modal ? (
    <ReactionViewModal {...props} />
  ) : (
    <ReactionView {...props} />
  );
};

export default RootReaction;
