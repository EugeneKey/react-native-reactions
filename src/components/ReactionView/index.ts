import RootReaction from './RootReaction';

export { RootReaction as Reaction };
export type { ReactionViewProps } from './types';
