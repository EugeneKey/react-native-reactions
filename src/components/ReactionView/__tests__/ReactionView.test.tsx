import React from 'react';
import { render } from '@testing-library/react-native';
import ReactionView from '../ReactionView';

jest.useFakeTimers();

describe('ReactionView component', () => {
  it('Match Snapshot', () => {
    const { toJSON } = render(<ReactionView items={[]} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
