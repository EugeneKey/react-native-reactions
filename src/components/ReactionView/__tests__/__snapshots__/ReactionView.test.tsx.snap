// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionsView component Match Snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    0
  </Text>
</RCTSafeAreaView>
`;
